### PR TITLE
Doc for distribution of analyzer config in nuget

### PR DESCRIPTION
**Pull request by jjolidon**
## Summary

This commit adds documentation for the distribution of `.globalconfig` files with nugets.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/code-analysis/configuration-files.md](https://github.com/dotnet/docs/blob/09cbc03c47ec8895c24b78c70218641e7cdb73cc/docs/fundamentals/code-analysis/configuration-files.md) | [Configuration files for code analysis rules](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files?branch=pr-en-us-35719) |

### CODE REVIEW

**Comment by jjolidon** [comment]
@dotnet-policy-service agree

**Comment by jjolidon** [comment]
The information comes from this comment https://github.com/dotnet/roslyn/issues/47707#issuecomment-1075412726 made by @jmarolf on Roslyn.

**Comment by jjolidon** [comment]
I fixed the hardcoded locale, but I don't know how to make a relative link to the documentation of nuget. Alternatively I could remove the link.

File: docs/fundamentals/code-analysis/configuration-files.md [hunk 1]
```diff
@@ -99,6 +99,17 @@ Consider the following naming recommendations:
 > [!NOTE]
 > The top-level entry `is_global = true` is not required when the file is named `.globalconfig`, but it is recommended for clarity.
 
+### Distribution in NuGet packages
+
+As mentionned above, `GlobalAnalyzerConfigFiles` can be distributed with NuGet packages. To do so, [add a `.props`](https://learn.microsoft.com/nuget/concepts/msbuild-props-and-targets) file to the nuget, and include the `GlobalAnalyzerConfigFiles` directly under the `Project` node, like so:
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/configuration-files.md:104]:
```suggestion
Global AnalyzerConfig files can be distributed with NuGet packages. To do so, add a [*.props* file](/nuget/concepts/msbuild-props-and-targets) to the NuGet package. In the *.props* file, add a `GlobalAnalyzerConfigFiles` item under the `Project` node:
```

**Comment by BillWagner** [comment]
@gewarren 

It looks like you reviewed this with one suggestion. Should we accept the suggestion and merge?

File: docs/fundamentals/code-analysis/configuration-files.md [hunk 2]
```diff
@@ -99,6 +99,17 @@ Consider the following naming recommendations:
 > [!NOTE]
 > The top-level entry `is_global = true` is not required when the file is named `.globalconfig`, but it is recommended for clarity.
 
+### Distribution in NuGet packages
+
+Global AnalyzerConfig files can be distributed with NuGet packages. To do so, add a [*.props* file](/nuget/concepts/msbuild-props-and-targets) to the NuGet package. In the *.props* file, add a `GlobalAnalyzerConfigFiles` item under the `Project` node:
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/configuration-files.md:104]:
```suggestion
Global AnalyzerConfig files can be distributed with NuGet packages. To do so, add a [*.props* file](/nuget/concepts/msbuild-props-and-targets) to the NuGet package. In the *.props* file, add a `GlobalAnalyzerConfigFiles` item under the `Project` node:

```

**Review by gewarren — APPROVED** [review]

**Pull request merged**